### PR TITLE
Replace boost::regex library with std::regex

### PR DESCRIPTION
**Pull request by djuanit0x**
### Description
I address #6971

### CODE REVIEW

File: libdevcore/Whiskers.cpp [hunk 1]
```diff
@@ -25,8 +25,6 @@
 
 #include <libdevcore/Assertions.h>
 
-#include <boost/regex.hpp>
```
Comment (chriseth) [inline libdevcore/Whiskers.cpp:28]: I think you have to replace this by an std include.

**Comment by chriseth** [comment]
There still seem to be some problems here. Did you try to compile locally?

**Comment by djuanit0x** [comment]
Nope, I keep getting an error when running `./scripts/tests.sh`. The Failing CI logs, however, consistently produce these errors `error: no matching function for call to ‘regex_replace`. After reading [this question](https://stackoverflow.com/questions/24990646/how-do-i-use-regex-replace), It appears that the compiler should be updated to GCC 4.9 to remove the errors.

**Comment by chriseth** [comment]
It looks like stdc++ does not support regex_replace where the replacement is performed by a lambda function. So you would have to write some glue code. This might be more complicated than I initially though - sorry about that!

**Comment by djuanit0x** [comment]
No problem:smile:. I am closing this PR, since I have some troubles with the testing.

**Comment by Xiad32** [comment]
Why are you guys saying regex_replace is not in std? I found this in the reference: 
https://en.cppreference.com/w/cpp/regex/regex_replace

**Comment by chriseth** [comment]
@djuanit0x thanks a lot for trying!

**Pull request closed without merging**